.github/workflows/validate.yaml

```diff
@@ -0,0 +1,39 @@
+name: Validate Definitions
+
+on:
+  push:
+    branches:
+      - main   # push commit to the main branch
+  pull_request:
+    branches:
+      - main   # pull request to the main branch
+
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  build-linux:
+    name: CI py${{ matrix.python-version }}
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.12"
+
+      - name: Install Requirements
+        run: |
+          python -m pip install --upgrade pip setuptools
+          python -m pip install -r requirements.txt
+
+      - name: validate
+        run: |
+          python -m dev_tools nxtest
```

Comment:
Does this check **all** NXDL files or just the nxtest.nxdl.xml file?

Reply:
all.

Reply:
Can additional logging be enabled that shows each NXDL file as it is being validated?  Not see this in the recent validation CI logs:
https://productionresultssa18.blob.core.windows.net/actions-results/a8a7d118-b555-4269-8bbd-a422a596a1d6/workflow-job-run-d5c11ce9-a08a-5bde-783f-1bfee628ff52/logs/job/job-logs.txt?rsct=text%2Fplain&se=2024-09-28T12%3A29%3A02Z&sig=%2B%2BsuRQU%2Bi1l4%2FTda7NTubMnBen85z2Y%2FWxVvNVLWgW4%3D&ske=2024-09-28T23%3A30%3A09Z&skoid=ca7593d4-ee42-46cd-af88-8b886a2f84eb&sks=b&skt=2024-09-28T11%3A30%3A09Z&sktid=398a6654-997b-47e9-b12b-9515b896b4de&skv=2024-08-04&sp=r&spr=https&sr=b&st=2024-09-28T12%3A18%3A57Z&sv=2024-08-04

Reply:
Looking at the [Quick Reference](https://github.com/nexusformat/definitions/tree/main/dev_tools#quick-reference), I'm looking for:

`pytest dev_tools`

Reply:
Locally, this command will test all NXDL files.

Reply:
If we stopped validating NXDL files on every push, that was a regression.

Reply:
```bash
(nexus_defs) prjemian@arf:~/.../NeXus/definitions$ python -m dev_tools nxtest
usage: __main__.py [-h] {nxclass,manual,impatient} ...
__main__.py: error: argument command: invalid choice: 'nxtest' (choose from 'nxclass', 'manual', 'impatient')
```

Reply:
OK, tested with your branch and confirm this tests all existing NXDL files.  It reported properly an error that I introduced into my local NXdata just to test the code.